src/Core/Components/Tabs/FluentTabs.razor.cs

```diff
@@ -154,41 +154,44 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
             _jsModuleOverflow = await JSRuntime.InvokeAsync<IJSObjectReference>("import", JAVASCRIPT_FILE.FormatCollocatedUrl(LibraryConfiguration));
 
             var horizontal = Orientation == Orientation.Horizontal;
-            await _jsModuleOverflow.InvokeVoidAsync("fluentOverflowInitialize", _dotNetHelper, Id, horizontal, FLUENT_TAB_TAG,25);
+            await _jsModuleOverflow.InvokeVoidAsync("fluentOverflowInitialize", _dotNetHelper, Id, horizontal, FLUENT_TAB_TAG, 25);
         }
     }
 
     private async Task HandleOnTabChangedAsync(TabChangeEventArgs args)
     {
         var tabId = args?.ActiveId;
-        if (tabId is not null && _tabs.TryGetValue(tabId, out FluentTab? tab))
+        var tab = _tabs.FirstOrDefault(i => i.Id == tabId);
+
+        if (tab is not null && _tabs.Contains(tab))
         {
             await OnTabChange.InvokeAsync(tab);
-            ActiveTabId = tabId;
+            ActiveTabId = tabId!;
             await ActiveTabIdChanged.InvokeAsync(tabId);
         }
     }
 
     internal int RegisterTab(FluentTab tab)
     {
-        _ = _tabs.TryAdd(tab.Id!, tab);
+        _tabs.Add(tab);
         return _tabs.Count - 1;
     }
 
-    internal async Task UnregisterTabAsync(FluentTab tab)
+    internal async Task UnregisterTabAsync(string id)
     {
         if (OnTabClose.HasDelegate)
         {
+            var tab = _tabs.FirstOrDefault(t => t.Id == id);
             await OnTabClose.InvokeAsync(tab);
         }
 
         if (_tabs.Count > 0)
         {
-            _tabs.Remove(tab.Id!);
+            _tabs.RemoveAt(_tabs.Count - 1);
```

Comment:
Why to remove the last one and not the Tab ID?

Reply:
This doesn't explain why calling the “UnregisterTabAsync(string id)” method removes the last item from the tab list :-)

Reply:
We've confirmed that for this fix, it is doing the right thing